helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1045,4 +1048,54 @@ private static void setNextJobPurgeTime(String workflow, long currentTime, long
       rebalanceScheduler.scheduleRebalance(manager, workflow, nextPurgeTime);
     }
   }
+
+  /**
+   * Get target thread pool size from InstanceConfig first; if that fails, get it from
+   * ClusterConfig; if that fails, fall back to the default value.
+   * @param configAccessor - accessor used for the configs
+   * @param clusterName - the cluster name for InstanceConfig and ClusterConfig
+   * @param instanceName - the instance name for InstanceConfig
+   * @return target thread pool size
+   */
+  public static int getTargetThreadPoolSize(ConfigAccessor configAccessor, String clusterName,
+      String instanceName) {
+    // Check instance config first for thread pool size
+    try {
+      InstanceConfig instanceConfig = configAccessor.getInstanceConfig(clusterName, instanceName);
+      if (instanceConfig != null) {
+        int targetTaskThreadPoolSize = instanceConfig.getTargetTaskThreadPoolSize();
+        if (targetTaskThreadPoolSize > 0) {
+          return targetTaskThreadPoolSize;
+        }
```

Comment:
nit, It would be helpful to note what happens if it is "else" condition.

Reply:
Ok, adding a comment.